### ISSUE TITLE
** does not match zero directories

### ISSUE THREAD

**Issue by houmain** [p-ranav/glob#3]
Hi, thank you for this nice library!

I think glob::rglob("dir/**/*.ext") should match:
- dir/file.ext
- dir/sub/file.ext
- dir/sub/sub/file.ext

Currently it only matches:
- dir/sub/file.ext
- dir/sub/sub/file.ext

Or how else one can recursively get all .ext files within dir?

**Comment by p-ranav** [comment]
Yes, I agree. Glob in this case should include `dir/file.ext`. Thanks for reporting!

**Comment by destructor64** [comment]
Is this issue fixed with the merge of #26 or still open? We are using this library in a project and in our testing, this still doesn't work.

If I have the following file tree:
```
$ tree test
test
|-- dir
|   |-- dir1
|   |   `-- test2.txt
|   `-- test1.txt
`-- test0.txt
```
and try to use the `rglob` function with the pattern `test/**/*.txt` on the folder `test`, it selects `test1.txt` and `test2.txt`, but **not** `test0.txt`
